src/devices/oil_watchman_advanced.c

```diff
@@ -0,0 +1,102 @@
+/** @file
+    Watchman Sonic Advanced/Plus oil tank level monitor
+    
+    Copyright (C) 2023 Gareth Potter
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+*/
+/**
+
+Tested devices:
+- Watchman Sonic Advanced
+*/
+#include "decoder.h"
+
+
+static int oil_watchman_advanced_callback(r_device *decoder, bitbuffer_t *bitbuffer)
+{
+    static const uint8_t PREAMBLE_SYNC_LENGTH_BITS = 40;
+    static const uint8_t MODEL_LENGTH_BITS = 24;
+    static const uint8_t BODY_LENGTH_BITS = 128;
+    // total length of message is 192 bits
+    // preamble is 40 bits of 10101... then the 'standard' sync 0x2dd4, then a model number, 0x0e0401
+    // no need to match all the preamble; 24 bits worth should do
+    uint8_t const match_pattern[8] = {0xaa, 0xaa, 0xaa, 0x2d, 0xd4, 0x0e, 0x04, 0x01};
+
+    uint8_t *b;
```

Comment:
declare most of these where you actually use them, e.g. line 63 `int depth = b[0];`

Reply:
Right, sorry about this: I had thought this was some sort of traditional C style.

Reply:
Well some of our code is still with all vars on top, but modern compilers complain about scoping ;)